contracts/src/lunarswap-v1/LunarswapLiquidity.compact

```diff
@@ -0,0 +1,238 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Midnight Apps Contracts v0.0.1-alpha.0 (lunarswap-v1/LunarswapLiquidity.compact)
+
+pragma language_version >= 0.18.0;
+
+/**
+ * @title LunarswapLiquidity
+ * @description Liquidity Provider (LP) token management system for the Lunarswap protocol.
+ * 
+ * @remarks
+ * The LunarswapLiquidity module manages the creation, minting, burning, and tracking of liquidity
+ * provider tokens. These tokens represent a user's share of liquidity in a trading pair and
+ * entitle holders to a portion of trading fees and the underlying assets.
+ * 
+ * Key Features:
+ * - LP token minting for liquidity providers
+ * - LP token burning for liquidity removal
+ * - Total supply tracking per trading pair
+ * - Token metadata management (name, symbol, decimals)
+ * - Nonce-based token generation
+ * - UTXO-based token model
+ */
+module LunarswapLiquidity {
+  import CompactStandardLibrary;
+
+  import "../../../compact-contracts/contracts/src/archive/ShieldedToken" prefix ShieldedToken_;
+  import "../math/Uint128" prefix Uint128_;
+  import "../math/Bytes32" prefix Bytes32_;
+
+  /**
+   * @ledger _counter
+   * @description
+   * Internal counter used for LP token nonce evolution and minting operations.
+   */
+  ledger _counter: Counter;
+
+  /**
+   * @ledger lpTokenNonce
+   * @description
+   * The global nonce used for LP token generation and uniqueness.
+   */
+  export ledger lpTokenNonce: Bytes<32>;
```

Comment:
This strikes me as odd. First, we're not prefixing this with an underscore like we are with `_counter`. Second, is there a reason to repeatedly use the `lpToken` prefix? Why not just `nonce` or `_nonce`?